tests/test_nsz_service.py

```
@@ -0,0 +1,299 @@
+    monkeypatch.setattr(nsz_module.nsz_service, "resolved_keys_file", lambda: None)
+
+    spawned = False
+
+    async def fake_exec(*_args, **_kwargs):
+        nonlocal spawned
+        spawned = True
+        return _FakeProcess(pid=1, chunks=[])
+
+    monkeypatch.setattr(nsz_module.asyncio, "create_subprocess_exec", fake_exec)
+
+    with pytest.raises(RuntimeError, match="prod.keys"):
```

Ruff (0.15.15)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)